src/main/java/org/folio/consortia/service/impl/SyncPrimaryAffiliationServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+import lombok.extern.log4j.Log4j2;
+
+import static org.folio.consortia.utils.TenantContextUtils.prepareContextForTenant;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class SyncPrimaryAffiliationServiceImpl implements SyncPrimaryAffiliationService {
+  private final UserService userService;
+  private final UserTenantService userTenantService;
+  private final TenantService tenantService;
+  private final UserTenantRepository userTenantRepository;
+  private final FolioModuleMetadata folioModuleMetadata;
+  private final FolioExecutionContext folioExecutionContext;
+  private final ObjectMapper objectMapper = new ObjectMapper();
+  private final KafkaService kafkaService;
+  private final SyncPrimaryAffiliationClient syncPrimaryAffiliationClient;
+
+  @Override
+  @Async("asyncTaskExecutor")
+  public void syncPrimaryAffiliations(UUID consortiumId, String tenantId) {
+    log.info("Start syncing user primary affiliations for tenant {}", tenantId);
+    List<User> users = new ArrayList<>();
+    try {
+      users = userService.getUsersByQuery("cql.allRecords=1", 0, Integer.MAX_VALUE);
+    } catch (Exception e) {
+      log.error("syncPrimaryAffiliations:: failed to retrieve '{}' users", tenantId, e);
+    }
+    if (CollectionUtils.isNotEmpty(users)) {
+      SyncPrimaryAffiliationBody spab = buildSyncPrimaryAffiliationBody(tenantId, users);
+      syncPrimaryAffiliationClient.savePrimaryAffiliations(spab, consortiumId.toString(), tenantId);
+    }
+  }
+
+  private SyncPrimaryAffiliationBody buildSyncPrimaryAffiliationBody(String tenantId, List<User> users) {
+    var syncUsers = users.stream()
+      .map(user -> new SyncUser().id(user.getId())
+        .username(user.getUsername()))
+      .toList();
+    return new SyncPrimaryAffiliationBody().tenantId(tenantId)
+      .users(syncUsers);
+  }
+
+  @Override
+  @Async("asyncTaskExecutor")
+  public void createPrimaryUserAffiliations(UUID consortiumId, SyncPrimaryAffiliationBody syncPrimaryAffiliationBody) {
+    FolioExecutionContext currentTenantContext = (FolioExecutionContext) folioExecutionContext.getInstance();
+    log.info("Start creating user primary affiliation for tenant {}", syncPrimaryAffiliationBody.getTenantId());
+    var tenantId = syncPrimaryAffiliationBody.getTenantId();
+    var userList = syncPrimaryAffiliationBody.getUsers();
+    var centralTenantId = tenantService.getCentralTenantId();
+    TenantEntity tenantEntity = tenantService.getByTenantId(tenantId);
+
+    try (var context = new FolioExecutionContextSetter(prepareContextForTenant(centralTenantId, folioModuleMetadata, currentTenantContext))) {
+      IntStream.range(0, userList.size())
+        .sequential()
+        .forEach(idx -> {
+          var user = userList.get(idx);
+          log.info("Processing users: {} of {}", idx + 1, userList.size());
+
+          var consortiaUserTenant = userTenantRepository.findByUserIdAndTenantId(UUID.fromString(user.getId()), tenantId)
```

Comment:
findByUserId instead of findByUserIdAndTenantId, if user with uuid is already affiliated somewhere, we don't need to try to primary affiliate it